addons/mrp_subcontracting/models/stock_picking.py

```diff
@@ -147,17 +147,33 @@ def _prepare_subcontract_mo_vals(self, subcontract_move, bom):
 
     def _subcontracted_produce(self, subcontract_details):
         self.ensure_one()
+
+        group_move = {}
+        group_by_company = defaultdict(list)
         for move, bom in subcontract_details:
             if float_compare(move.product_qty, 0, precision_rounding=move.product_uom.rounding) <= 0:
                 # If a subcontracted amount is decreased, don't create a MO that would be for a negative value.
                 continue
-            mo = self.env['mrp.production'].with_company(move.company_id).create(self._prepare_subcontract_mo_vals(move, bom))
-            self.env['stock.move'].create(mo._get_moves_raw_values())
-            self.env['stock.move'].create(mo._get_moves_finished_values())
-            mo.date_planned_finished = move.date  # Avoid to have the picking late depending of the MO
-            mo.action_confirm()
+            mo_subcontract = self._prepare_subcontract_mo_vals(move, bom)
+            mo_subcontract['date_planned_finished'] = move.date  # Avoid to have the picking late depending on the MO
```

Comment:
if you create the finished stock move after that we will loose this update
 https://github.com/odoo/odoo/blob/2f97805b7416fe36c2f9b6df44e7ae3dd8e4569d/addons/mrp/models/mrp_production.py#L759

Reply:
@Whenrow So it can be better to add the date in the `finished_move.write` to remove the additional write after the mo is created since I don't see any other change that needs to be taken into account from the `write` of `MrpProduction`. But, indeed, I didn't see that.

Reply:
@auma-odoo I just checked, and it seems we don't need to rewrite the date after the creation anymore. The line was introduced in 326ae2a725715, but commit 94cb5b6e8f9f9 makes it useless.
You can remove it :)

Reply:
@auma-odoo Correct

Reply:
@Whenrow It seems that I need to keep this line because it makes a test fail without it